doc/luasnip.txt

```diff
@@ -1453,6 +1454,152 @@ callback will have access to the `POSTFIX_MATCH` field as well.
 <
 
 
+SNIPPET LIST                                            *luasnip-snippet-list*
+
+>
+    local sl = require("luasnip.extras.snippet_list")
+<
+
+
+Makes an `open` function available to use to open currently available snippets
+in a different buffer/window/tab.
+
+`sl.open(opts:table|nil)`
+
+
+- `opts`: optional arguments:
+    - `snip_info`: `snip_info(snippet) -> table representation of snippet`
+    - `printer`: `printer(snippets:table) -> string representation of snippets`
+    - `display`: `display(snippets:string)`
```

Comment:
I think we should ignore that `printer` returns a string by default, and allow it to return anything. (we might want to have structured data in `display` when building some kind of tree, and then `printer` would not be necessary (e.g., it would be the identity-function))

Reply:
Hmm this is interesting because by default we're kinda counting on `printer` returning a string. So the only scenario where printer could NOT return a string would be if the user implemented their own `display` as well.

So maybe I should add another section explaining that by default `display` is called with the result from `printer` and if you want to return a table that has a certain structure from `printer` or hell return anything you'd have to implement your own `display` that parses that input OR like you said you could just implement your own display and do whatever you want in there without even using printer.

Hmm what do you think?

Reply:
I've been thinking along the same line(is there a reason for separating display and printer), but I think it makes sense to separate them since the most common return type of printer is a string, and then all displays accepting a string are compatible (otherwise, there'd be some duplicate code in the displays)
I think going into more detail on this would be great👍👍

Reply:
Is this https://github.com/L3MON4D3/LuaSnip/pull/652/commits/74f508ff91e54f530d7b795807587c366c4ab0e4 sufficient you think?

Reply:
Nice! Maybe parse -> stringify, in the last line (at least I think that's more like what you meant, am I right?)

Reply:
Yeah that's better! I fixed it. If that's everything, let me know and I'll squash the commits and separate it into just two different commits (Code Implementation and Documentation)

Reply:
Okay awesome, this is it :partying_face::partying_face:
Two works :ok_hand: